drivers/modbus/Kconfig

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2023-2024 Bas Stottelaar <basstottelaar@gmail.com>
```

Comment:
We are no longer doing dependency modelling with kconfig... This can be dropped.

Reply:
Right, so I can just remove the Kconfig file?

Reply:
Removed the Kconfig file.